src/hypofuzz/dashboard.py

```diff
@@ -247,31 +246,52 @@ def linear_elapsed_time(self, *, since: float = -math.inf) -> list[float]:
 
 
 def test_for_websocket(test: Test) -> dict[str, Any]:
-    # only send necessary attributes for the dashboard
-    test = dataclasses.asdict(test)
-    # report.worker is duplicated among all reports in a reports_by_worker list.
-    # drop them and add them as an attribute to reports_by_workers?
-    # so reports_by_workers: dict[str, tuple[WorkerIdentity, list[Report]]]
-    del test["rolling_observations"]
-    del test["corpus_observations"]
-    del test["linear_reports"]
-    test["reports_by_worker"] = {
-        worker_uuid: [report_for_websocket(report) for report in reports]
-        for worker_uuid, reports in test["reports_by_worker"].items()
+    # Limit to 1000 reports per test. If we're over 1000 total reports, sample
+    # evenly from each worker.
+    #
+    # As a temporary stopgap until a more intelligent algorithm, sample the
+    # entirety of the first 20 reports per worker, then `::step` after. This
+    # helps with (4).
```

Comment:
If we want to downsample a lot, can we keep new behaviors and most recent report (for each worker) only?

I'd really like to avoid downsampling this harshly, but I guess maybe we have to...

Reply:
quick math: 300 bytes per report, 10 tests, 1000 fingerprints per test = 300 * 10 * 1000 = 3000000 bytes = 3 mb. (and that's up to 30 if you have 100 tests etc).

Agree we want more intelligent sampling here, and maybe we can reduce this after optimizing the report format (msgpack etc) or the d3 plotting code.